network-api/networkapi/wagtailpages/tests/buyersguide/test_products.py

```diff
@@ -121,6 +121,189 @@ def test_localized_related_products_non_default_locale(self):
         # If there is no localized version, the default version should be returned:
         self.assertIn(second_product, related_products_fr)
 
+    def test_preview_related_products(self):
+        """
+        Tests if `preview_related_products` accurately orders and includes related products for page previews.
+        """
+        product_page = self.product_page
+
+        related_product_1 = buyersguide_factories.ProductPageFactory(parent=self.bg)
+        related_product_2 = buyersguide_factories.ProductPageFactory(parent=self.bg)
+        related_product_3 = buyersguide_factories.ProductPageFactory(parent=self.bg)
+
+        buyersguide_factories.RelatedProductsFactory(
+            page=product_page,
+            related_product=related_product_1,
+            sort_order=2,
+        )
+        buyersguide_factories.RelatedProductsFactory(
+            page=product_page,
+            related_product=related_product_2,
+            sort_order=0,
+        )
+        buyersguide_factories.RelatedProductsFactory(
+            page=product_page,
+            related_product=related_product_3,
+            sort_order=1,
+        )
+
+        related_preview_products = product_page.preview_related_products
+
+        self.assertEqual(len(related_preview_products), 3)
+        # Order should be product 2, product 3, product 1.
+        self.assertListEqual(
```

Comment:
Would it be better to use `assertCountEqual`?

Reply:
Hey Jhonatan! I actually removed asserts relating to count in my latest commit. I will resolve this conversation for now, but can reopen/add more updates if we choose to re-implement them.